From 1d3a464a2e7db2eaa6e63fa806cea45bff3365b2 Mon Sep 17 00:00:00 2001
From: Gyorgy Sarvari <skandigraun@gmail.com>
Date: Thu, 9 Dec 2021 20:24:38 +0100
Subject: [PATCH] Fixed inspectdb.tests.InspectDBTestCase.test_custom_fields()
 on SQLite 3.37+.

From: Mariusz Felisiak <felisiak.mariusz@gmail.com>

Use FlexibleFieldLookupDict which is case-insensitive mapping because
SQLite 3.37+ returns some data type names upper-cased e.g. TEXT.

Upstream-Status: Backport [https://github.com/django/django/commit/974e3b8750fe96c16c9c0b115a72ee4a2171df34]
Signed-off-by: Gyorgy Sarvari <skandigraun@gmail.com>
---
 tests/inspectdb/tests.py | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/tests/inspectdb/tests.py b/tests/inspectdb/tests.py
index bb5e457..a2519cf 100644
--- a/tests/inspectdb/tests.py
+++ b/tests/inspectdb/tests.py
@@ -256,18 +256,17 @@ class InspectDBTestCase(TestCase):
         Introspection of columns with a custom field (#21090)
         """
         out = StringIO()
-        orig_data_types_reverse = connection.introspection.data_types_reverse
-        try:
-            connection.introspection.data_types_reverse = {
+        with mock.patch(
+            'django.db.connection.introspection.data_types_reverse.base_data_types_reverse',
+            {
                 'text': 'myfields.TextField',
                 'bigint': 'BigIntegerField',
-            }
+            },
+        ):
             call_command('inspectdb', 'inspectdb_columntypes', stdout=out)
             output = out.getvalue()
             self.assertIn("text_field = myfields.TextField()", output)
             self.assertIn("big_int_field = models.BigIntegerField()", output)
-        finally:
-            connection.introspection.data_types_reverse = orig_data_types_reverse
 
     def test_introspection_errors(self):
         """
